vendor/src/github.com/docker/engine-api/types/types.go

```diff
@@ -274,6 +274,19 @@ type ExecStartCheck struct {
 	Tty bool
 }
 
+type HealthcheckResult struct {
+	Start, End time.Time
+	ExitCode   int    // 0=healthy, 1=unhealthy, 2=starting
+	Output     string // Output from last check
```

Comment:
Why would it round-trip? If the script generates invalid UTF-8 then Go will replace the bad data with unicode replacement characters. The probe output is intended to be displayed as text, not used as some kind of reliable binary transport.